docs/src/pages/blog/nextjs-root-params.mdx

```diff
@@ -0,0 +1,343 @@
+
+But first, let's have a look at how this API works in practice.
+
+## Root layouts
+
+Previously, Next.js required a [root layout](https://nextjs.org/docs/app/api-reference/file-conventions/layout#root-layouts) to be present at `app/layout.tsx`.
+
+Now, you can move these to a nested folder that can be a dynamic segment, e.g.:
+
+```
+src
+└── app
+    └── [locale]
+        ├── layout.tsx (root layout)
+        └── page.tsx
+```
+
+A root layout is a layout that has no other layouts located above it.
+
+In contrast, layouts that do have other layout ancestors are regular layouts:
+
+```
+src
+└── app
+    └── [locale]
+        ├── layout.tsx
+        ├── (...)
+        └── news
+            ├── layout.tsx (regular layout)
+            └── page.tsx
+```
+
+With the addition of `rootParams`, you can now read the param values of a root layout in all Server Components that render within it:
+
+```tsx filename=src/components/LocaleSwitcher.tsx
+import {unstable_rootParams as rootParams} from 'next/server';
+
+export async function LocaleSwitcher() {
+  // Read the value of `[locale]`
+  const {locale} = await rootParams();
+
+  // ...
+}
+```
+
+## Multiple root layouts
+
+Here's where it gets interesting: With [route groups](https://nextjs.org/docs/app/building-your-application/routing/route-groups), you can provide another layout for pages that are not located in the `[locale]` segment:
+
+```
+src
+└── app
+    ├── [locale]
+    │   ├── layout.tsx
+    │   └── page.tsx
+    └── (unlocalized)
+        ├── layout.tsx
+        └── page.tsx
+```
+
+The layout at `[locale]/layout.tsx` as well as the layout at `(unlocalized)/layout.tsx` both have no other layouts located above them, therefore both qualify as root layouts. Due to this, in this case the returned value of `rootParams` will depend on where the component that calls the function is being rendered from.
+
+If you call `rootParams` in shared code that is used by both root layouts, this allows for a pattern like this:
+
+```tsx filename="src/utils/getLocale.tsx"
+import {unstable_rootParams as rootParams} from 'next/server';
+
+export async function getLocale() {
+  // Try to read the locale in case we're in `[locale]/layout.tsx`
+  let {locale} = await rootParams();
+
+  // If we're in `(unlocalized)/layout.tsx`, let's use a fallback
+  if (!locale) {
+    locale = 'en';
+  }
+
+  return locale;
+}
+```
+
+With this, you can use the `getLocale` function across your codebase to read the current locale without having to worry about where it's being called from.
+
+In an internationalized app, this can for example be useful to implement a country selection page at the root where you have to rely on a default locale.
```

Comment:
```suggestion
In an internationalized app, this can for example be useful to implement a country selection page at the root where you have to rely on a default locale. Once the user is within the `[locale]` segment, this param value can be used instead for localizing page content.
```